src/hotspot/share/logging/logSelectionList.cpp

```diff
@@ -90,6 +90,10 @@ bool LogSelectionList::parse(const char* str, outputStream* errstream) {
       break;
     }
     _selections[_nselections++] = selection;
+
+    if (comma_pos == nullptr) {
+      break;
+    }
```

Comment:
I think this check is not necessary here since it is checked in the for loop condition already - or am I overlooking something?

Reply:
Unfortunately we still run into the issue , even with the existing loop condition.
check the flowchart of the for loop 
https://www.programiz.com/cpp-programming/for-loop
the update condition is done after the body ; then we run into the issue.

Reply:
Ah, right, I see. But in that case, we can/should remove the check for comma_pos != nullptr from the condition because we're explicitly doing it in the loop body then.

Reply:
Agree, the check can be simplified .